doc/introduction/transforming_circuits.rst

```diff
@@ -0,0 +1,100 @@
+.. role:: html(raw)
+   :format: html
+
+.. _intro_ref_transform_circuits:
+
+Transforming circuits
+=====================
+
+A quantum transform is a crucial concept in PennyLane, it is represented by a function that takes a circuit as input and
+yields one or more transformed circuits along with a post-processing function. The post-processing function is applied
+to the results obtained after executing the transformed circuit(s). This becomes particularly valuable when a transform
+generates multiple circuits, requiring a method to aggregate the results (e.g. parameter shift or Hamiltonian expansion).
+
+In PennyLane, these requirements are translated as follows:
+
+* A transform accepts a :class:`~.QuantumTape` as its primary input and
+  returns a sequence of :class:`~.QuantumTape` and an associated processing function.
+
+To streamline the creation of transforms and ensure their versatility across various circuit abstractions in PennyLane
+(:class:`~.QuantumTape`, :class:`~.QNode` and quantum functions), we have created a simple
+decorator :func:`pennylane.transform` that can be applied on your quantum transform respecting the above contract.
+
+Creating your own transform
+---------------------------
+
+To illustrate the process of creating a quantum transform, let's consider a straightforward example. Suppose we want
+a transform that removes all :class:`~.RX` operations from a given circuit. In this case, we merely need to filter the
+original :class:`~.QuantumTape` and return a new one without the filtered operations. As we don't require a specific processing
+function in this scenario, we include a null function that simply returns the result.
+
+.. code-block:: python
+
+    from typing import Sequence, Callable
+    from pennylane.tape import QuantumTape
+
+    def remove_rx(tape: QuantumTape) -> (Sequence[QuantumTape], Callable):
+
+        operations = filter(lambda op: op.name != "RX", tape.operations)
+        new_tape = type(tape)(operations, tape.measurements, shots=tape.shots)
+
+        def null_postprocessing(results):
+            return results[0]
```

Comment:
Why do you have to index? For me, this isn't a null function, I would expect `lambda *args: args` to be a null function 🤔